src/canvasObject.js

```diff
@@ -1,23 +1,41 @@
 'use strict';
 
 require('openseadragon');
+var ImageResource = require('./ImageResource');
 
 var CanvasObject = function(config, dispatcher) {
   this.fullyOpened = config.fullyOpened || false;
-  this.visible = config.visible || true; // todo: this is not used yet. Do we need it?
   this.clipRegion = config.clipRegion;
   this.opacity = config.opacity || 1;
-  this.x = config.x || 0;
-  this.y = config.y || 0;
-  this.placeholder = config.placeholder || { type: 'image', url: './example-thumbnail.png' };
   this.index = config.index;
 
   this.id = config.canvas['@id'];
-  this.height = config.canvas.height;
-  this.width = config.canvas.width;
-  this.images = config.canvas.images;
+
+  this.bounds = {
+    x : config.x || 0,
+    y : config.y || 0,
+    height : config.canvas.height,
+    width : config.canvas.width
+  };
+
+  // details and alternates possibly go here; disambiguate between them.
+  this.images = config.canvas.images.map(function(image) {
+    return new ImageResource(
+      {
+        tileSource: image.resource.service['@id'] + '/info.json'
+      },
+      dispatcher
+    );
+  });
+
+  this.thumbnail = new ImageResource(
+  {
+    tileSource: config.canvas.thumbnail || this.images[0].tileSource
```

Comment:
In that demo, we request a 2X sized thumbnail in case we need to use the resizing method. It also ensures that we have the resolution we need as we zoom down into the object from thumbnail view.